support/oidc-discovery-provider/handler.go

```diff
@@ -53,35 +69,49 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var host string
-	var path string
-	var urlScheme string
-	if h.jwtIssuer != "" {
-		jwtIssuerURL, _ := url.Parse(h.jwtIssuer)
-		host = jwtIssuerURL.Host
-		path = jwtIssuerURL.Path
-		urlScheme = jwtIssuerURL.Scheme
+	var issuerURL *url.URL
+	if h.jwtIssuer != nil {
+		issuerURL = &url.URL{
+			Scheme: h.jwtIssuer.Scheme,
+			Host:   h.jwtIssuer.Host,
+			Path:   h.jwtIssuer.Path,
+		}
 	} else {
-		host = r.Host
-		urlScheme = "https"
+		urlScheme := "https"
 		if h.allowInsecureScheme && r.TLS == nil && r.URL.Scheme != "https" {
 			urlScheme = "http"
 		}
+		issuerURL = &url.URL{
+			Scheme: urlScheme,
+			Host:   r.Host,
+		}
 	}
 
-	if err := h.verifyHost(host); err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
+	var jwksURI *url.URL
+	if h.jwksURI != nil {
+		jwksURI = h.jwksURI
+	} else {
+		tmpURLScheme := "https"
+		if h.allowInsecureScheme && r.TLS == nil && r.URL.Scheme != "https" {
+			tmpURLScheme = "http"
+		}
```

Comment:
This part is common to the two so we could move it outside of the two ifs (for issuer and jwks urls).